### ISSUE TITLE
Cypress tests with Okta Authentication issues

### ISSUE THREAD

**Issue by gemmie** [cypress-io/cypress#18216]
### Current behavior

While running Cypress tests with Okta Authentication, after successfully mocking logging in or implementing Okta Authentication according to Cypress documentation, the token in localStorage gets overwritten to `{}` - which causes logging out and tests failing.

In addition this behaviour is very inconsistent, as once in a dozen of tries the test pass. 

### Desired behavior

localStorage is not overwritten, tests can safely log in.
Behavior is consistent - logs in every time.

### Test code to reproduce

`localStorage.setItem('okta-token-storage', value)`

### Cypress Version

7.7.0

**Comment by jennifer-shehane** [comment]
Have you tried using [`cy.session`](https://on.cypress.io/session)? It's experimental but meant to help with juggling things like localStorage around.

**Comment by fsee85** [comment]
Did you solve the issue?
We have similar issues while using signInWithCredentials method from okta-auth-js api while in cypress browser.
Basically the error response is fullfilled with undefined error causes.

**Comment by gemmie** [comment]
Hi! We did not solve the issue with storage being overwritten per se. We changed our implementation to match the one in cypress docs as mentioned in my last comment. The issue we had with that was we had `client_id` in envs, which worked locally but failed in travis tests. We had to assign the `client_id` from `process.env.OKTA_CLIENT_ID` to the cypress envs. As once Cypress runs it apparently does not see the regular env.
